sys-apps/shadow/shadow-4.11.1.ebuild

```diff
@@ -1,19 +1,20 @@
-# Copyright 1999-2020 Gentoo Authors
+# Copyright 1999-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
 
 TMPFILES_OPTIONAL=1
-inherit autotools libtool pam systemd tmpfiles
+inherit libtool pam systemd tmpfiles
 
 DESCRIPTION="Utilities to deal with user accounts"
 HOMEPAGE="https://github.com/shadow-maint/shadow"
-SRC_URI="https://github.com/shadow-maint/shadow/releases/download/${PV}/${P}.tar.xz"
+SRC_URI="https://github.com/shadow-maint/shadow/releases/download/v${PV}/${P}.tar.xz"
 
 LICENSE="BSD GPL-2"
-SLOT="0"
-KEYWORDS="~alpha amd64 arm arm64 hppa ~ia64 ~m68k ~mips ppc ppc64 ~riscv s390 sparc x86"
-IUSE="acl audit bcrypt cracklib nls pam selinux skey split-usr +su xattr"
+# Subslot is for libsubid's SONAME.
+SLOT="0/4"
+KEYWORDS="~alpha amd64 arm arm64 hppa ~ia64 ~m68k ~mips ppc ppc64 ~riscv ~s390 sparc x86"
+IUSE="acl audit bcrypt cracklib nls pam selinux skey split-usr su xattr"
```

Comment:
Just a heads up:
1. They broke ABI (so remember to add := to consumers and revbump them, if they use libsubuid -- not many of these exist. Most things just use shadow binaries, so not need this).
2. We changed default from +su -> su and now use util-linux to provide su. You may want to re-enable su for shadow and disable it for util-linux instead. This is because shadow was deprecating it entirely but now will be keeping it for non-PAM users.

Hope those hints might be useful somehow.

Reply:
Thanks for the heads-up.

> They broke ABI (so remember to add := to consumers and revbump them, if they use libsubuid -- not many of these exist. Most things just use shadow binaries, so not need this). IIRC this is just podman and maybe one or two other container-related packages.

Currently no binary in the Flatcar production images is linking against libsubid.
(If we could somehow add podman, then it might be a different story, but not in the short future.)
For now I think there is no need to worry.

> We changed default from +su -> su and now use util-linux to provide su. You may want to re-enable su for shadow and disable it for util-linux instead if you want to avoid churn. This is because shadow was deprecating it entirely but now will be keeping it for non-PAM users. But my recommendation is to use util-linux su as that is the preference of shadow upstream too.

That's an interesting point.
I agree that Flatcar should also use su in util-linux instead of shadow.

However, it looks like, changes like that seem to open up a can of worms.
First, under the `/build/amd64-usr`, it looks ok to rely on su from util-linux.
OTOH, in the SDK, installation of `util-linux[pam,su]` requires `pam_sss.so` to be installed in the SDK, because baselayout of Flatcar requires that.
I am not sure it is a good idea to add the whole sssd into the SDK, only for the pam_sss module. :thinking: